### PR TITLE
Allow query parameters in `with_request_url`

### PR DESCRIPTION
**Pull request by javierm**
### Summary

Query parameters were ignored when using the `with_request_url` test helper. With these changes, that's no longer the case :smile:.

### Other Information

My knowledge of the internals of Rails routing is very limited, so if there's a more elegant solution than using `set_header` and `parse_query`, please let me know! :wink:

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ title: Changelog
 
 ## main
 
+* Allow query parameters in `with_request_url`
```
Comment (joelhawksley) [inline docs/CHANGELOG.md:10]:
```suggestion
* Allow query parameters in `with_request_url` test helper.
```

File: docs/guide/testing.md [hunk 2]
```diff
@@ -137,6 +137,19 @@ class ExampleComponentTest < ViewComponent::TestCase
 end
 ```
 
+You can also set `request.query_parameters`, using the same method:
```
Comment (joelhawksley) [inline docs/guide/testing.md:140]: It's also possible to set query parameters:

File: lib/view_component/test_helpers.rb [hunk 3]
```diff
@@ -73,12 +73,15 @@ def with_controller_class(klass)
 
     def with_request_url(path)
       old_request_path_parameters = request.path_parameters
+      old_request_query_parameters = request.query_parameters
       old_controller = defined?(@controller) && @controller
 
       request.path_parameters = Rails.application.routes.recognize_path(path)
+      request.set_header("action_dispatch.request.query_parameters", request.send(:parse_query, path.split("?")[1]))
```
Comment (joelhawksley) [inline lib/view_component/test_helpers.rb:80]: Did you look into a way to pull this off with a public API? I'd really prefer we avoid using `send`.

**Review by joelhawksley — COMMENTED** [review]
👋🏻 looking good! Thank you for the nice tests and docs ❤️

File: lib/view_component/test_helpers.rb (hunk 3 above)
Comment (javierm) [reply lib/view_component/test_helpers.rb:80]: I agree 100% :+1:. The other day I didn't find anything and hoped there was a way to set query parameters similar to `#path_parameters=` that most people knew about.

I'll have another look :+1:.

**Comment by javierm** [comment]
@joelhawksley Thank you very much for your review! :pray:

I've [applied your suggestions](https://github.com/github/view_component/compare/14bcf013e..96bc25f69). Also added another assertion for the case of several query parameters, to make sure the `&` is correctly treated as the separator.

If you've got any more comments, they're welcome :wink:.

**Review by joelhawksley — APPROVED** [review]

**Comment by joelhawksley** [comment]
Thank you for your contribution, Javi!

**Pull request merged**